packages/sil-governance/src/routes/dashboard/components/ProposalsList.tsx

```
@@ -34,16 +44,80 @@ const getFilter = (filterType: ElectionFilter, currentUser: Address | null): Pro
+  if (proposal1.vote === undefined) return 1;
+  if (proposal2.vote === undefined) return -1;
+  return proposal1.vote - proposal2.vote;
+};
+
+interface Props {
+  readonly filterType: ElectionFilter;
 }
 
 const ProposalsList = ({ filterType }: Props): JSX.Element => {
-  const proposals = ReactRedux.useSelector((state: RootState) => state.proposals.proposals);
-  const governor = ReactRedux.useSelector((state: RootState) => state.extension.governor);
-  const blockchain = ReactRedux.useSelector((state: RootState) => state.blockchain);
+  const proposals = useSelector((state: RootState) => state.proposals.proposals);
+  const governor = useSelector((state: RootState) => state.extension.governor);
+  const blockchain = useSelector((state: RootState) => state.blockchain);
+
+  const [comparatorId, setComparatorId] = useState<ComparatorId>(ComparatorId.IdDescending);
+
+  const changeComparator = (selectedItem: SelectFieldItem): void => {
+    switch (selectedItem.name) {
+      case ComparatorLabels.ExpiryDate:
+        setComparatorId(ComparatorId.ExpiryDate);
+        break;
+      case ComparatorLabels.StartDate:
+        setComparatorId(ComparatorId.StartDate);
+        break;
+      case ComparatorLabels.Vote:
+        setComparatorId(ComparatorId.Vote);
+        break;
+      default:
+        setComparatorId(ComparatorId.IdDescending);
+    }
+  };
+
+  const { form } = useForm({ onSubmit: () => {} });
 
   const filter = getFilter(filterType, governor ? governor.address : null);
+
+  let comparator: ProposalsComparator;
+  switch (comparatorId) {
+    case ComparatorId.IdDescending:
+      comparator = compareByIdDescending;
+      break;
+    case ComparatorId.ExpiryDate:
+      comparator = compareByExpiryDate;
+      break;
+    case ComparatorId.StartDate:
+      comparator = compareByStartDate;
+      break;
+    case ComparatorId.Vote:
+      comparator = compareByVote;
+      break;
+    default:
+      comparator = compareByIdDescending;
```

Can you throw an exception here? This only happens if a new value is added but no switch case is created. In this case we want hard crashed to detect this programming error early.